draft-ietf-avtcore-rtp-over-quic.md

```diff
@@ -563,11 +573,17 @@ stream credits it will have to provide to the media-sending peer.
 ## QUIC Datagrams {#quic-datagrams}
 
 Senders can also transmit RTP packets in QUIC datagrams. QUIC datagrams are an
-extension to QUIC described in {{!RFC9221}}. QUIC datagrams preserve frame
-boundaries. Thus, a single RTP packet can be mapped to a single QUIC datagram
-without additional framing. Senders SHOULD consider the header overhead
-associated with QUIC datagrams and ensure that the RTP/RTCP packets, including
-their payloads, flow identifier, QUIC, and IP headers, will fit into path MTU.
+extension to QUIC described in {{!RFC9221}}. QUIC datagrams can only be used if
+the use of the extension was successfully negotiated during the QUIC handshake.
+If the use of datagrams was signalled using a signalling protocol, but the
+datagram extension was not negotiated during the handshake, a peer MAY close the
+connection with the ROQ\_SIGNALING\_ERROR error code.
```

Comment:
Hmmm. RFC 9000 says this:

> An endpoint MUST treat the receipt of a frame of unknown type as a connection error of type FRAME_ENCODING_ERROR.

We just had a conversation about whether a "frame of known type defined for an extension that has not been negotiated" would be treated as "a frame of unknown type" or some other kind of error, but the high order bit is that I don't think the RTP receiver will ever see that datagram - the QUIC receiver will have already punted it as a connection error.

Reply:
I think we are talking about two different error cases. One problem is a peer sending a datagram frame in a connection that did not negotiate the use of QUIC datagrams. In that case, I agree that the QUIC receiver will react first, and the RTP receiver will never know about it. The case I intended to cover is that the two peers used some out-of-band signaling before setting up the QUIC connection, and in that signaling, they agreed to use QUIC datagrams, but then during the QUIC handshake, one of the peers does not send the transport parameter to negotiate datagrams. In that QUIC connection, datagrams are not allowed, so the peer that was expecting to be allowed to use datagrams can immediately close the connection.

Reply:
@mengelbart's explanation makes sense to me